.github/workflows/code_quality.yml

```
@@ -0,0 +1,36 @@
+name: Code Quality
+on: [pull_request, create]
```

```suggestion
on:
  push:
    branches:
      - main
  pull_request:
  create:
```

We should at least run this on push to `main`, i.e. after any PR is merged.